src/contracts/middleware/IndexRegistry.sol

```diff
@@ -0,0 +1,188 @@
+        return uint32(quorumToOperatorList[quorumNumber].length);
+    }
+
+
+    /// @notice Returns the current number of operators of this service.
+    function totalOperators() external view returns (uint32){
+        return uint32(globalOperatorList.length);
+    }
+
+
+    function _updateTotalOperatorHistory(uint8 quorumNumber) internal {
+
+        //if there is a prior entry, update its "toBlockNumber"
+        if (totalOperatorsHistory[quorumNumber].length > 0) {
+            totalOperatorsHistory[quorumNumber][totalOperatorsHistory[quorumNumber].length - 1].toBlockNumber = uint32(block.number);
+        }
+
+        OperatorIndex memory totalOperatorUpdate;
+        // In the case of totalOperatorsHistory, the index parameter is the number of operators in the quorum
+        totalOperatorUpdate.index = uint32(quorumToOperatorList[quorumNumber].length);
+        totalOperatorsHistory[quorumNumber].push(totalOperatorUpdate);
+    }
+
+    /// 
+    /// @param operatorId operatorId of the operator to update
+    /// @param quorumNumber quorumNumber of the operator to update
+    /// @param index the latest index of that operator in quorumToOperatorList
+    function _updateOperatorIdToIndexHistory(bytes32 operatorId, uint8 quorumNumber, uint32 index) internal {
+        uint256 operatorIdToIndexHistoryLength = operatorIdToIndexHistory[operatorId][quorumNumber].length;
+
+        //if there is a prior entry for the operator, set the previous entry's toBlocknumber
+        if (operatorIdToIndexHistoryLength > 0) {
+            operatorIdToIndexHistory[operatorId][quorumNumber][operatorIdToIndexHistoryLength - 1].toBlockNumber = uint32(block.number);
+        }
+        OperatorIndex memory latestOperatorIndex;
+        latestOperatorIndex.index = index;
+        operatorIdToIndexHistory[operatorId][quorumNumber].push(latestOperatorIndex);
+    }
+
+    /// @notice when we remove an operator from quorumToOperatorList, we swap the last operator in 
+    ///         quorumToOperatorList[quorumNumber] to the position of the operator we are removing
+    /// @param quorumNumber quorum number of the operator to remove
+    /// @param indexToRemove index of the operator to remove
+    function _removeOperatorFromQuorumToOperatorList(uint8 quorumNumber, uint32 indexToRemove) internal {   
+        
+        uint32 quorumToOperatorListLastIndex  = uint32(quorumToOperatorList[quorumNumber].length - 1);
+
+        // if the operator is not the last in the list, we must swap the last operator into their positon
+        if(indexToRemove != quorumToOperatorListLastIndex){
+            bytes32 operatorIdToSwap = quorumToOperatorList[quorumNumber][quorumToOperatorListLastIndex];
+            //update the swapped operator's operatorIdToIndexHistory list with a new entry, as their index has now changed
+            _updateOperatorIdToIndexHistory(operatorIdToSwap, quorumNumber, indexToRemove);
```

Comment:
we also need to update the removed operator's index history by setting the `toBlockNumber` to the current block.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/blob/a0381d062e2d64bd8d45088b6f2281b3b99ed01d/src/contracts/middleware/IndexRegistry.sol#L68

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/blob/42234da18a78b710b1fa9a695820d4a6c06edf9e/src/contracts/middleware/IndexRegistry.sol#L83